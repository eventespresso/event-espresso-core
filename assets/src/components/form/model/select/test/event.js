import EventSelect, { EditorEventSelect } from '../event';
import { shallow, render } from 'enzyme';

jest.mock('../../base/model-select', () => () => <span>EventSelect</span>);

describe('EventSelect()', () => {
	it('matches snapshot with default props', () => {
		const wrapper = shallow(<EventSelect />);
		expect(wrapper).toMatchSnapshot();
	});
	it('matches expected default props', () => {
		const wrapper = shallow(<EventSelect />);
		expect(wrapper.prop('queryData')).toEqual({
			limit: 100,
			order: 'DESC',
			orderBy: 'start_date',
			showExpired: false,
		});
		expect(wrapper.prop('selectConfiguration').placeholder).toEqual(
			'Select Eventâ€¦'
		);
		expect(wrapper.prop('modelName')).toEqual('event');
	});
	it('does not allow overriding the modelName', () => {
		const wrapper = shallow(<EventSelect modelName={'tickets'} />);
		expect(wrapper.prop('modelName')).toEqual('event');
	});
});

describe('EditorEventSelect()', () => {
	it('matches snapshot with default props', () => {
		const wrapper = render(<EditorEventSelect />);
		expect(wrapper).toMatchSnapshot();
	});
});
