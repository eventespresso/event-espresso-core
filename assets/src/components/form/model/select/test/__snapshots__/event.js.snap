// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditorEventSelect() matches snapshot with default props 1`] = `
<div
  class="components-base-control css-wdf2ti-Wrapper e1puf3u0"
>
  <div
    class="components-base-control__field css-11vcxb9-StyledField e1puf3u1"
  >
    <label
      class="components-base-control__label css-pezhm9-StyledLabel e1puf3u2"
      for="inspector-select-event-control-0"
    >
      Select Event
    </label>
    <span>
      EventSelect
    </span>
  </div>
</div>
`;

exports[`EventSelect() matches snapshot with default props 1`] = `
<Component
  getQueryString={[Function]}
  label="Select Event"
  modelName="event"
  optionsEntityMap={
    Object {
      "default": Object {
        "label": "EVT_name",
        "value": "EVT_ID",
      },
    }
  }
  queryData={
    Object {
      "limit": 100,
      "order": "DESC",
      "orderBy": "start_date",
      "showExpired": false,
    }
  }
  selectConfiguration={
    Object {
      "defaultValue": undefined,
      "loadingMessage": [Function],
      "noOptionsMessage": [Function],
      "onChange": undefined,
      "placeholder": "Select Eventâ€¦",
    }
  }
/>
`;
